### ISSUE TITLE
Negative loss when pre-training on CC12M

### ISSUE THREAD

**Issue by ivonajdenkoska** [Optimization-AI/fast_clip#2]
Hi,

Thanks a lot for sharing the code. I'm trying to reproduce the pre-training on CC12M using 4 H100 but I'm getting a negative loss after training for a while (see the screenshot). Have you also observed this? Thanks in advance!

![image](https://github.com/user-attachments/assets/36b4a569-d09e-41c3-85ce-503793ce89f6)

**Comment by xywei00** [comment]
Yes, this is normal. In short, this "loss" is only a term that facilitates computing the gradient estimator in compositional optimization, and it is not the value of the loss function.
